### PR TITLE
move aie initialization to when loading xclbin

### PR DESCRIPTION
**Pull request by larry9523**
1) Move AieArray initialization to the time after we load xcblin.
2) Replace term Cardano

### CODE REVIEW

File: src/runtime_src/core/edge/user/shim.h [hunk 1]
```diff
@@ -156,7 +157,7 @@ class shim {
   int xclLog(xrtLogMsgLevel level, const char* tag, const char* format, ...);
 
 #ifdef XRT_ENABLE_AIE
-  zynqaie::Aie *aieArray;
+  std::shared_ptr<zynqaie::Aie> aieArray;
```
Comment (stsoe) [inline src/runtime_src/core/edge/user/shim.h:160]: What the lifetime of this object?  Is really shared ownership?  Or should it be owned by the shim object and die when the shim object is deleted?  I think it's unique ownership in which case it should be std::unique_ptr and getAieArray() should return the raw pointer.   This doesn't change functionality, just communicates intent.

Comment (larry9523) [reply src/runtime_src/core/edge/user/shim.h]: Thanks Soren. With the returning raw pointer suggestion, I can make getAieArray() works with unique pointer.

File: src/runtime_src/core/edge/user/shim.cpp [hunk 2]
```diff
@@ -1398,11 +1393,17 @@ getAieArray()
...
 {
-  aieArray = aie;
+  aieArray = std::make_shared<zynqaie::Aie>(mCoreDevice);
 }
 
+bool
+shim::
+isAieRegisted()
+{
+  return !(aieArray.get() == nullptr);
```
[the remark this answers is not in the document]
Comment (larry9523) [reply src/runtime_src/core/edge/user/shim.cpp]: changed to (aieArray != nullptr)

File: src/runtime_src/core/edge/user/shim.cpp [hunk 3]
```diff
@@ -1398,11 +1393,17 @@ getAieArray()
 
 void
 shim::
-setAieArray(zynqaie::Aie *aie)
+registAieArray()
```
[the remark this answers is not in the document]
Comment (larry9523) [reply src/runtime_src/core/edge/user/shim.cpp]: done

**Pull request merged**